STM/vertx/src/main/java/demo/demo12/TheatreVerticle.java

```diff
@@ -0,0 +1,118 @@
+package demo.demo12;
+
+import demo.domain.Booking;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+import com.arjuna.ats.arjuna.AtomicAction;
+
+import demo.domain.TheatreService;
+import demo.domain.ServiceResult;
+
+public abstract class TheatreVerticle extends AbstractVerticle {
+    private TheatreService serviceClone;
+
+    public TheatreVerticle() {
+    }
+
+    abstract TheatreService getClone();
+
+    @Override
+    public void start(Future<Void> future) throws Exception {
+        serviceClone = getClone();
+
+        startServer(future, config().getInteger("port"));
+    }
+
+    private void startServer(Future<Void> future, int listenerPort) {
+        Router router = Router.router(vertx);
+
+        getRoutes(router);
+
+        // Create the HTTP server and pass the "accept" method to the request handler.
+        vertx
+                .createHttpServer()
+                .requestHandler(router::accept)
+                .listen(listenerPort,
+                        result -> {
+                            if (result.succeeded()) {
+                                future.complete(); // tell the caller the server is ready
+                            } else {
+                                result.cause().printStackTrace(System.out);
+                                future.fail(result.cause()); // tell the caller that server failed to start
+                            }
+                        }
+                );
+
+        assert router.getRoutes().size() > 0;
+
+        String route1 = router.getRoutes().get(0).getPath();
+
+        System.out.printf("%s service listening on http://localhost:%d%s%n" , getServiceName(), listenerPort, route1);
+    }
+
+    private void getRoutes(Router router) {
+        router.get(String.format("/api/%s", getServiceName())).handler(this::getBookings);
+        router.post(String.format("/api/%s/:seats", getServiceName())).handler(this::makeBooking);
+    }
+
+    private String getServiceName() {
+        return "theatre";
+    }
+
+    private void getBookings(RoutingContext routingContext) {
```

Comment:
I like it - it looks good to me. I still think it would aid readability to move the business logic methods to a separate class compared the Vert.x bootstrapping but that shouldn't prevent a merge. If you are ready I will approve this